server/src/main/java/org/elasticsearch/action/admin/cluster/node/stats/NodeStats.java

```diff
@@ -240,6 +254,11 @@ public void readFrom(StreamInput in) throws IOException {
         } else {
             adaptiveSelectionStats = null;
         }
+        if (in.getVersion().onOrAfter(Version.V_6_5_0) && in.readBoolean()) {
+            shardsStats = in.readList(ShardStats::readShardStats);
+        } else {
+            shardsStats = null;
+        }
```

Comment:
More for my own education: why do we need the version check here?

Reply:
The transport layer needs to be safe and compatible across different nodes. For example, during a rolling upgrade this code may be talking to a node that runs any version before it. In that case, they wouldn't know what to do with the mess of data at the end of this stream and would assume that something was wrong.

Reply:
Ah yes, the rolling upgrade case, of course! Thanks.